### PR TITLE
Added unittest for callbacks

### PR DESCRIPTION
**Pull request by leandropineda**
### Description

Added unittest for callback registering and refactored old unittests

### CODE REVIEW

File: inorbit_edge/tests/test_robot_session_callbacks.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    # callback gets registered
+    robot_session._on_connect(..., ..., ..., 0)
+
+    assert my_command_handler in robot_session.command_callbacks
+    robot_session.client.subscribe.assert_called_with(
+        topic="r/id_123/custom_command/script/command"
+    )
+
+
+def test_robot_session_callback_on_message(mocker, mock_mqtt_client, mock_inorbit_api):
```
Comment (miguelgarcia) [inline inorbit_edge/tests/test_robot_session_callbacks.py:50]: What is this test's goal ? It seems to test that we comply with the InOrbit protocol and Echo messages are sent in response to commands, but it does not check that `my_command_handler` is called.

**Review by miguelgarcia — CHANGES_REQUESTED** [review]
Hey @leandropineda I think that the tests verify some internal conditions, like `subscribe` being called or some details about our protocol, but I believe the tests should assert that when a callback is registered, it's then actually called with the right values. To do that we may need to extend or improve the MQTT mock.

WDYT ?

File: inorbit_edge/tests/test_robot_session_callbacks.py (hunk 1 above)
Comment (leandropineda) [reply inorbit_edge/tests/test_robot_session_callbacks.py:50]: You are right. I created a new test just for testing the Echo message, and extended other test to verify that the handler was called

File: inorbit_edge/tests/test_robot_session_callbacks.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    # connect robot_session so it populates properties with API response data
+    robot_session.connect()
+    # manually execute on_connect callback so the ``custom_command_callback``
+    # callback gets registered
+    robot_session._on_connect(..., ..., ..., 0)
+
+    assert my_command_handler in robot_session.command_callbacks
+    robot_session.client.subscribe.assert_called_with(
+        topic="r/id_123/custom_command/script/command"
+    )
```
Comment (miguelgarcia) [inline inorbit_edge/tests/test_robot_session_callbacks.py:50]: isn't this already implicitly covered by `test_robot_session_executes_command_callback_on_message` ? I have the non-strong feeling that checking this internal behavior in the tests isn't necessary or a good idea.

**Review by miguelgarcia — APPROVED** [review]
LGTM, I leave to you the decision about keeping or removing the tests that just check internal behavior

File: inorbit_edge/tests/test_robot_session_callbacks.py (hunk 2 above)
Comment (leandropineda) [reply inorbit_edge/tests/test_robot_session_callbacks.py:50]: No, they are different tests. This one only checks that the SDK subscribes to the custom command topic when a callback is registered. `test_robot_session_executes_command_callback_on_message` checks that the callbacks are executed with specific parameters

**Pull request merged**